src/VisualStudio/Core/Impl/RoslynVisualStudioWorkspace.cs

```diff
@@ -49,17 +51,21 @@ internal class RoslynVisualStudioWorkspace : VisualStudioWorkspaceImpl
 
         internal override IInvisibleEditor OpenInvisibleEditor(DocumentId documentId)
         {
-            var globalUndoService = this.Services.GetService<IGlobalUndoService>();
+            var globalUndoService = this.Services.GetRequiredService<IGlobalUndoService>();
             var needsUndoDisabled = false;
 
+            var textDocument = this.CurrentSolution.GetTextDocument(documentId);
+
+            Contract.ThrowIfNull(textDocument, string.Format(WorkspacesResources._0_is_not_part_of_the_workspace, documentId));
```

Comment:
>Contract.ThrowIfNull(textDocument, string.Format(WorkspacesResources._0_is_not_part_of_the_workspace, documentId)); [](start = 12, length = 115)

Let's not localize here - this will actually fetch the resource and format the string and then throw away the result in common case.

Reply:
Fixed.